src/types.rs

```diff
@@ -365,7 +365,7 @@ impl Types {
 
         match &self.0 {
             Dyn() | Num() | Bool() | Str() | Var(_) => true,
-            Array(ty) if ty.0 == AbsType::Dyn() => true,
+            Flat(rt) if matches!(*rt.term, Term::Var(_)) => true,
```

Comment:
Oh, did you fix a but, while you were at it?

Reply:
The removal is just consequence of removing the alias, but yes the second line is a change to a tiny detail I noticed before. When printing an `Array` type, unneeded parentheses were added when the parameter was a contract, like `Array (Foo)` instead of `Array Foo`, which is fixed here.